src/transformers/models/layoutlmv2/feature_extraction_layoutlmv2.py

```diff
@@ -220,3 +233,54 @@ def __call__(
             encoded_inputs["boxes"] = boxes_batch
 
         return encoded_inputs
+
+    def post_process(self, outputs, target_sizes, offset_mapping, bbox):
+        """
+        Converts the output of :class:`~transformers.LayoutLMv2ForTokenClassification` into the format expected by the COCO api.
+        Only supports PyTorch.
+
+        Args:
+            outputs :obj:`Dict`):
+                Raw outputs of the model.
+            target_sizes (:obj:`torch.Tensor` of shape :obj:`(batch_size, 2)`):
+                Tensor containing the size (h, w) of each image of the batch. For evaluation, this must be the original
+                image size (before any data augmentation). For visualization, this should be the image size after data
+                augment, but before padding.
+            offset_mapping (:obj:`torch.Tensor` of shape :obj:`(batch_size, x, 2)`):
+                TTensor coming that is "offset_mapping" field of outputs from
+                :class:`~transformer.LayoutLMv2TokenizerFast`.
+            bbox (:obj:`torch.Tensor` of shape :obj:`(batch_size, x, 4)`):
+                Tensor coming that is "bbox" field of outputs from :class:`~transformer.LayoutLMv2TokenizerFast`.
+
+        Returns:
+            :obj:`List[Dict]`: A list of dictionaries, each dictionary containing the scores, labels and boxes for an
+            image in the batch as predicted by the model.
+        """
+        out_logits = outputs.logits
+
+        assert len(out_logits) == len(
+            target_sizes
+        ), "Make sure that you pass in as many target sizes as the batch dimension of the logits"
+        assert (
+            target_sizes.shape[1] == 2
+        ), "Each element of target_sizes must contain the size (h, w) of each image of the batch"
+
+        prob = nn.functional.softmax(out_logits, -1)
+        scores, labels = prob[..., :-1].max(-1)
+        scores = scores.tolist()
+        labels = labels.tolist()
+        boxes = bbox.tolist()
+        target_sizes = target_sizes.tolist()
+        is_subword = np.array(offset_mapping.tolist())[:, :, 0] != 0
```

Comment:
Did some refactoring and added comments. Wdyt
It works and was used in the test_method below 😊
https://github.com/huggingface/transformers/blob/64c2df5570ff97024d60500570055ff44361f142/src/transformers/models/layoutlmv2/feature_extraction_layoutlmv2.py#L274-L281

Reply:
It's much better. Looking at the actual data, I understood it's linked to some captured text being multi-token, and we're taking only the first token.

Few things:
- `subword` is not an adapted language IMO. subwords don't necessarily exist for random tokenizers so we should stay away from the term `word` if possible. Here you're talking about "first token". `is_first_token`, and `offsets`.
- `offsets` are offsets, so let's keep their name. IMO `offset_start == 0 # in case of multiple tokens we look only at the beginning of the string` seems much easier to understand. (You do have to know what an offset is, but I think it's easier)

Reply:
And maybe let the door open to having more complex behavior. We're dropping a lot of info here. It's perfectly fine in this PR, but future readers/contributors might want to improve this.

Reply:
changed the variable names as suggested 👍